### PR TITLE
CH4/OFI: added FI_COMPLETION flag to rma_read call

### PR DESCRIPTION
**Pull request by hoopoepg**
- there is an issue in function MPIDI_OFI_do_rdma_read: it
  prepare request assuming that completion event is generated
  on operation completed. But EP is initialized by flag
  FI_SELECTIVE_COMPLETION which prevents CQ entry by default.
  Fix: use fi_readmsg(..., FI_COMPLETION) call

### CODE REVIEW

**Comment by raffenet** [comment]
test:jenkins/ch4/ofi

File: src/mpid/ch4/netmod/ofi/ofi_am_events.h [hunk 1]
```diff
@@ -152,13 +152,31 @@ static inline int MPIDI_OFI_do_rdma_read(void *dst,
...
         MPIDI_OFI_cntr_incr();
-        MPIDI_OFI_CALL_RETRY_AM(fi_read(MPIDI_OFI_EP_TX_RMA(0),
-                                        (char *) dst + done,
-                                        curr_len, NULL,
-                                        MPIDI_OFI_comm_to_phys(comm, src_rank, MPIDI_OFI_API_TAG),
-                                        src + done,
-                                        MPIDI_OFI_AMREQUEST_HDR(rreq, lmt_info).rma_key,
-                                        &am_req->context), FALSE /* no lock */ , read);
+
+        struct iovec iov = {
```
Comment (raffenet) [inline src/mpid/ch4/netmod/ofi/ofi_am_events.h]: I realize now I already accepted a patch that used typed initialization like this, but we haven't actually discussed using it in CH4. IMO it's a good thing to have and am willing to make it a build requirement. We may have to do a little homework to see the current (and past) status of compiler support. @pavanbalaji any thoughts?

Comment (wesbland) [inline src/mpid/ch4/netmod/ofi/ofi_am_events.h]: Good point. I guess we can confirm that it works with GCC > 4.8 and ICC. 😄

Comment (hoopoepg) [inline src/mpid/ch4/netmod/ofi/ofi_am_events.h]: It's C99 standard, and it is supported by most of Unix/Linux compilers released after 2012.
such initialization is one of the best innovation in C99 :)

Comment (raffenet) [inline src/mpid/ch4/netmod/ofi/ofi_am_events.h]: I'm going to merge this and make an issue to review usage of designated initializers before we cut the next release.

**Pull request merged**